libres/lib/analysis/enkf_linalg.cpp

```diff
@@ -14,54 +14,15 @@
  * Implements Eq. 14.31 in the book Data Assimilation,
  * The Ensemble Kalman Filter, 2nd Edition by Geir Evensen.
 */
-void enkf_linalg_genX3(matrix_type *X3, const matrix_type *W,
-                       const matrix_type *D, const double *eig) {
-    const int nrobs = matrix_get_rows(D);
-    const int nrens = matrix_get_columns(D);
-    const int nrmin = std::min(nrobs, nrens);
-    int i, j;
-    matrix_type *X1 = matrix_alloc(nrmin, nrobs);
-    matrix_type *X2 = matrix_alloc(nrmin, nrens);
-
-    /* X1 = (I + Lambda1)^(-1) * W'*/
-    for (i = 0; i < nrmin; i++)
-        for (j = 0; j < nrobs; j++)
-            matrix_iset(X1, i, j, eig[i] * matrix_iget(W, j, i));
-
-    *X2 = *X1 * *D;
-    *X3 = *W * *X2;
-
-    matrix_free(X1);
-    matrix_free(X2);
-}
+Eigen::MatrixXd enkf_linalg_genX3(const Eigen::MatrixXd &W,
```

Comment:
Is this name descriptive of what happens?

Reply:
I think we can make do without most of the functions in `enkf_linalg.cpp` once all are refactored to use Eigen.
Right now, I don't have a better name for this function, but if you do I'll gladly rename it.

Reply:
I was hoping you might have a better alternative, as I dont, so no need to change it